test/e2e/network/dual_stack.go

```diff
@@ -686,6 +688,83 @@ var _ = common.SIGDescribe("[Feature:IPv6DualStack]", func() {
 	})
 })
 
+var _ = common.SIGDescribe("Dual Stack [Feature:PodHostIPs]", func() {
+	f := framework.NewDefaultFramework("dualstack")
+
+	ginkgo.It("should create pod, add host ips is empty", func() {
+		e2eskipper.SkipIfFeatureGateEnabled(kubefeatures.PodHostIPs)
+
+		podName := "pod-dualstack-host-ips"
+
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"test": "dualstack-host-ips"},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "dualstack-host-ips",
+						Image: imageutils.GetE2EImage(imageutils.Agnhost),
+					},
+				},
+			},
+		}
+
+		ginkgo.By("submitting the pod to kubernetes")
+		podClient := f.PodClient()
+		p := podClient.CreateSync(pod)
+
+		gomega.Expect(p.Status.HostIP).ShouldNot(gomega.BeEquivalentTo(""))
+		gomega.Expect(p.Status.HostIPs).Should(gomega.BeNil())
+
+		ginkgo.By("deleting the pod")
+		err := podClient.Delete(context.TODO(), pod.Name, *metav1.NewDeleteOptions(30))
+		framework.ExpectNoError(err, "failed to delete pod")
+	})
+
+	ginkgo.It("should create pod, add ipv6 and ipv4 ip to host ips", func() {
+		e2eskipper.SkipUnlessFeatureGateEnabled(kubefeatures.PodHostIPs)
+
+		podName := "pod-dualstack-host-ips"
+
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"test": "dualstack-host-ips"},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "dualstack-host-ips",
+						Image: imageutils.GetE2EImage(imageutils.Agnhost),
+					},
+				},
+			},
+		}
+
+		ginkgo.By("submitting the pod to kubernetes")
+		podClient := f.PodClient()
+		p := podClient.CreateSync(pod)
+
+		gomega.Expect(p.Status.HostIP).ShouldNot(gomega.BeEquivalentTo(""))
+		gomega.Expect(p.Status.HostIPs).ShouldNot(gomega.BeNil())
+
+		// validate first ip in HostIPs is same as HostIP
+		framework.ExpectEqual(p.Status.HostIP, p.Status.HostIPs[0].IP)
+		if len(p.Status.HostIPs) > 1 {
+			// assert 2 host ips belong to different families
+			if netutils.IsIPv4String(p.Status.HostIPs[0].IP) == netutils.IsIPv4String(p.Status.HostIPs[1].IP) {
```

Comment:
Is this expected? the docs on apis does not state any dual stackness req.

Reply:
Yes, It is reference to podIPs

https://github.com/kubernetes/kubernetes/blob/debd461d03b0643ebcc34dc818c2f274d11b8817/test/e2e/network/dual_stack.go#L56-L71